packages/number-input/src/NumberInput/NumberInput.spec.tsx

```diff
@@ -0,0 +1,390 @@
+    test('does not have basic accessibility issues', async () => {
+      const { container } = renderNumberInput({ label, ...defaultProps });
+      const results = await axe(container);
+      expect(results).toHaveNoViolations();
+    });
+  });
+
+  describe('unitless number input with default props', () => {
+    test('renders type as "number', async () => {
+      const { numberInput } = renderNumberInput({ label, ...defaultProps });
+      expect(numberInput.getAttribute('type')).toBe('number');
+    });
+
+    test('renders label container with correct text', () => {
+      const { labelElement } = renderNumberInput({ label, ...defaultProps });
+      expect(labelElement?.innerHTML).toContain(label);
+    });
+
+    test("doesn't render label container when label is not supplied", () => {
+      const { queryByRole } = renderNumberInput({
+        ['aria-label']: label,
+        ...defaultProps,
+      });
+      expect(queryByRole('label')).not.toBeInTheDocument();
+    });
+
+    test('renders description with correct text', () => {
+      const { queryByText } = renderNumberInput({
+        description,
+        ...defaultProps,
+      });
+      const descriptionEl = queryByText(description);
+      expect(descriptionEl).not.toBeNull();
+      expect(descriptionEl).toBeInTheDocument();
+    });
+
+    test("doesn't render description container when description is not supplied", () => {
+      const { queryByText } = renderNumberInput({ ...defaultProps });
+      expect(queryByText(description)).not.toBeInTheDocument();
+    });
+
+    test(`renders placeholder text`, () => {
+      const { getByPlaceholderText } = renderNumberInput({
+        label,
+        ...defaultProps,
+      });
+      expect(getByPlaceholderText(defaultProps.placeholder)).toBeVisible();
+    });
+
+    describe('when the "state" is "error"', () => {
+      test('renders warning icon ', () => {
+        const { container } = renderNumberInput({
+          state: State.Error,
+          ...defaultProps,
+        });
+        expect(
+          container.querySelector('svg[aria-label="Warning Icon"]'),
+        ).toBeInTheDocument();
+      });
+
+      test('renders error message', () => {
+        const { queryByText } = renderNumberInput({
+          label,
+          state: State.Error,
+          errorMessage,
+          ...defaultProps,
+        });
+        const errorEl = queryByText(errorMessage);
+        expect(errorEl).not.toBeNull();
+        expect(errorEl).toBeInTheDocument();
+      });
+    });
+
+    test('value change triggers onChange callback', () => {
+      const { numberInput } = renderNumberInput({
+        label,
+        ...defaultProps,
+      });
+
+      expect((numberInput as HTMLInputElement).value).toBe('');
+
+      fireEvent.change(numberInput, {
+        target: { value: '1' },
+      });
+
+      expect((numberInput as HTMLInputElement).value).toBe('1');
+      expect(defaultProps.onChange).toHaveBeenCalledTimes(1);
```

Comment:
should we test this with different values?

Reply:
i'll add a couple more